keylime/measured_boot.py

```diff
@@ -67,17 +69,26 @@ def get_policy(mb_refstate_str):
 
 
 def evaluate_policy(
-    mb_policy, mb_policy_name, mb_refstate_data, mb_measurement_data, pcrsInQuote, pcrPrefix, agent_id
+    mb_policy: Optional[Policy],
+    mb_policy_name: str,
+    mb_refstate_data: RefState,
+    mb_measurement_data: str,
+    pcrsInQuote: Set[int],
+    pcrPrefix: str,
+    agent_id: str,
 ) -> Failure:
     failure = Failure(Component.MEASURED_BOOT)
     missing = list(set(config.MEASUREDBOOT_PCRS).difference(pcrsInQuote))
     if len(missing) > 0:
         logger.error("%sPCRs specified for measured boot not in quote: %s", pcrPrefix, missing)
         failure.add_event("missing_pcrs", {"context": "PCRs are missing in quote", "data": missing}, True)
-    try:
-        reason = mb_policy.evaluate(mb_refstate_data, mb_measurement_data)
-    except Exception as exn:
-        reason = f"policy evaluation failed: {str(exn)}"
+    if mb_policy is None:
+        reason = "policy evaluation failed: missing measured boot policy"
+    else:
+        try:
+            reason = mb_policy.evaluate(mb_refstate_data, mb_measurement_data)
+        except Exception as exn:
```

Comment:
... maybe this would have dealt with trying to access None on evaluate(). I don't know.

Obviously the other choice would be to instrument the single call site to not enter this with None and change the above annotation to `Policy` from `Optional[Policy]`

Reply:
@maugustosilva  -- any suggestion?